src/mintpy/spatial_filter.py

```diff
@@ -50,12 +50,26 @@ def filter_data(data, filter_type, filter_par=None):
         data_filt = data - lp_data
 
     elif filter_type == "lowpass_gaussian":
-        data_filt = filters.gaussian(data, sigma=filter_par)
+        # Kai: watch out NaN holes (https://stackoverflow.com/questions/18697532/gaussian-filtering-a-image-with-nan-in-python)
+        V=np.array(data)
+        V[np.isnan(data)]=0
+        VV=filters.gaussian(V, sigma=filter_par)
+
+        W=np.ones_like(data)
+        W[np.isnan(data)]=0
+        WW=filters.gaussian(W, sigma=filter_par)
+
+        data_filt = VV/WW
+        data_filt[np.isnan(data)] = np.nan
+        #data_filt = filters.gaussian(data, sigma=filter_par)
```

Comment:
My test result was done on a data file that has been masked by `mask.py`. So the holes are filled with nan (1st image). 

The 2nd image is filtered by the original `skimage.filters.gaussian()` without handling the nan holes.

The 3rd image is, filling the nan holes with zero and low-pass filter on that (get field A). You can see the NW coastal region has velocity tapering down along the whole coast. Which is not true in the real data (1st image). Because we fill with zeros, then filter, so we have this artifact. 

The 4th image is this PR. Result = (filed A) / (field B). Where field B is a low-pass filtered auxiliary field. This auxiliary field is composed of ones/zeros, where zeros are the locations of the original nan holes.

"The idea is that replacing the NaNs by zeros introduces an error in the filtered array which can, however, be compensated by applying the same Gaussian filter to another auxiliary array and combining the two."

<img width="1361" alt="image" src="https://github.com/insarlab/MintPy/assets/55262505/6b5abf30-8fc0-46d9-8d21-5005f0de5b46">

Reply:
Got it. Nice and simple implementation! It's similar to the rubber sheeting that we talked about. Thank you @yuankailiu for the explanation.